sdk/servicebus/azure-servicebus/tests/perf_tests/T1_legacy_tests/_test_base.py

```diff
@@ -0,0 +1,148 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import uuid
+from urllib.parse import urlparse
+
+from azure_devtools.perfstress_tests import PerfStressTest, get_random_bytes
+
+from azure.servicebus import ServiceBusClient, ReceiveSettleMode, Message
+from azure.servicebus.aio import ServiceBusClient as AsyncServiceBusClient
+from azure.servicebus.control_client import ServiceBusService
+from azure.servicebus.control_client.models import Queue
+
+
+def parse_connection_string(conn_str):
+    conn_settings = [s.split("=", 1) for s in conn_str.split(";")]
+    conn_settings = dict(conn_settings)
+    shared_access_key = conn_settings.get('SharedAccessKey')
+    shared_access_key_name = conn_settings.get('SharedAccessKeyName')
+    endpoint = conn_settings.get('Endpoint')
+    parsed = urlparse(endpoint.rstrip('/'))
+    namespace = parsed.netloc.strip().split('.')[0]
+    return {
+        'namespace': namespace,
+        'endpoint': endpoint,
+        'entity_path': conn_settings.get('EntityPath'),
+        'shared_access_key_name': shared_access_key_name,
+        'shared_access_key': shared_access_key
+    }
+
+
+class _ServiceTest(PerfStressTest):
+    service_client = None
+    async_service_client = None
+
+    def __init__(self, arguments):
+        super().__init__(arguments)
+
+        connection_string = self.get_from_env("AZURE_SERVICEBUS_CONNECTION_STRING")
+        if not _ServiceTest.service_client or self.args.service_client_per_instance:
+            _ServiceTest.service_client = ServiceBusClient.from_connection_string(connection_string)
+            _ServiceTest.async_service_client = AsyncServiceBusClient.from_connection_string(connection_string)
+
+        self.service_client = _ServiceTest.service_client
+        self.async_service_client =_ServiceTest.async_service_client
```

Comment:
I should admit that it took me a while to understand how it works, and I found it's compact but perfect.

My only concern here is that in the case of `self.args.service_client_per_instance`, we keep updating the class variable `_ServiceTest.service_client`, would that affect the same class `_ServiceTest` used for `not self.args.service_client_per_instance` case -- or test running are isolated, updating class variable is okay

Reply:
You're right in that it's not overly clean - but I don't think it's going to cause problems. As soon as we assign to `self` on lines 46/47 then we're using local instance attribute.
However you're right that this is probably not best practice - I will clean it up a bit...